crates/storage/nippy-jar/src/writer.rs

```diff
@@ -113,7 +113,6 @@ impl<H: NippyJarHeader> NippyJarWriter<H> {
 
         // First byte of the offset file is the size of one offset in bytes
         offsets_file.write_all(&[OFFSET_SIZE_BYTES as u8])?;
-        offsets_file.sync_all()?;
```

Comment:
the idea behind this was: if the file is createad, then we should make sure the file pushes it to disk. can we keep some of that logic?

Reply:
the current logic is following https://github.com/paradigmxyz/reth/blob/23a72d3fce655b35f97428677cf43d2f3fd3c9ae/crates/storage/nippy-jar/src/writer.rs#L107-L117

and we can crash in between any of these lines, so we can end up with an empty file without any offsets written. My PR doesn't change this logic.